File: 35-search-insert-position/35-search-insert-position.cpp
class Solution {
public:
    int searchInsert(vector<int>& nums, int target) {
        if(nums.size() == 0) return 0;
        if(nums[nums.size()-1]<target)return nums.size();
		
		int l = 0, r = nums.size() - 1;
		while(l < r){
			int m = l + (r - l)/2;
			
			if(nums[m] == target) return m;
			else if(nums[m] > target) r = m;
			else l = m + 1;
		}
		
		return l;
    }
};

File: 35-search-insert-position/35-search-insert-position.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        l , r = 0, len(nums)-1
        while l <= r:
            mid=(l+r)//2
            if nums[mid] < target:
                l = mid+1
            else:
                if nums[mid]== target and nums[mid-1]!=target:
                    return mid
                else:
                    r = mid-1
        return l